flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/FlinkService.java

```diff
@@ -140,8 +141,13 @@ public boolean isJobManagerPortReady(Configuration config) {
         final String clusterId = config.get(KubernetesConfigOptions.CLUSTER_ID);
         final String namespace = config.get(KubernetesConfigOptions.NAMESPACE);
         final int port = config.getInteger(RestOptions.PORT);
-        final String host =
+        String host =
                 ExternalServiceDecorator.getNamespacedExternalServiceName(clusterId, namespace);
+        if (EnvUtils.get("KUBERNETES_SERVICE_HOST") == null) {
+            // not running in k8s, simplify local development
+            // TODO: make this an operator config and pass that config through
```

Comment:
If we are planning to make this configurable I suggest we do it straight away. Let's not leave TODOs for such minor additions.

Reply:
Haha yes, that is actually what I'm planning to do if the change as such is acceptable. Will have to pass `DefaultConfig` for that, not just the effective config for Flink. But that is probably a change worthwhile doing anyways as we are likely to add more settings that govern the operator behavior going forward.